### PR TITLE
Updated Twitter SDK [ New Version-1.2.0 ]

### PR DESCRIPTION
**Pull request by NitinBindal-Hevo**
Added functionality in SDK to get Job Metrics of particular account which will give us information whether we can schedule a new job on Twitter-Ads account or not and eventually avoid TwitterException as Twitter keeps changing the limit on No. of jobs that can in processing state and to make our code tolerate such changes and not fail this Job Metrics is needed.

Jira Ticket --> [INT-735](https://hevodata.atlassian.net/browse/INT-735)

Doc Link --> https://developer.twitter.com/en/docs/twitter-ads-api/analytics/api-reference/asynchronous

### CODE REVIEW

File: src/twitter4jads/util/TwitterAdHttpUtils.java [hunk 1]
```diff
@@ -64,4 +65,21 @@ public static RateLimitStatusImpl createFromResponseHeaderForCostBasedRateLimit(
 
         return new RateLimitStatusImpl(remainingHits, limit, resetTimeInSeconds, costValue);
     }
-}
+
+    public static JobLimitStatusImpl getJobLimitDataFromResponseHeader(HttpResponse res) {
+        int limit = 0;
+        int remaining = 0;
+
+        String strLimit = res.getResponseHeader("x-concurrent-job-limit");
```
Comment (adithya-1) [inline src/twitter4jads/util/TwitterAdHttpUtils.java:73]: Header field  `x-concurrent-job-limit` can be constant

File: src/twitter4jads/util/TwitterAdHttpUtils.java [hunk 2]
```diff
@@ -64,4 +65,21 @@ public static RateLimitStatusImpl createFromResponseHeaderForCostBasedRateLimit(
...
+        if(strLimit !=null) {
+            limit = Integer.parseInt(strLimit);
+        }
+
+        String strRemaining = res.getResponseHeader("x-concurrent-job-limit-remaining");
```
Comment (adithya-1) [inline src/twitter4jads/util/TwitterAdHttpUtils.java:78]: Same for `x-concurrent-job-limit-remaining`, can be a constant

File: src/twitter4jads/util/TwitterAdHttpUtils.java (hunk 1 above)
Comment (NitinBindal-Hevo) [reply src/twitter4jads/util/TwitterAdHttpUtils.java:73]: Are you saying something like this -->

const String someName = "x-concurrent-job-limit";

Comment (adithya-1) [reply src/twitter4jads/util/TwitterAdHttpUtils.java:73]: Yes, there is a class called TwitterAdsConstant that contains other constants like parameters required, so I thought to maintain uniformity we can add this as well there

Comment (adithya-1) [reply src/twitter4jads/util/TwitterAdHttpUtils.java:73]: Somthing like  `String RESPONSE_HEADER_CONCURRENT_JOB_LIMIT="x-concurrent-job-limit";`

Comment (adithya-1) [reply src/twitter4jads/util/TwitterAdHttpUtils.java:73]: Oh yes got it

Comment (adithya-1) [inline src/twitter4jads/util/TwitterAdHttpUtils.java:73]: One more doubt if it is case insensitive, like while reading api I saw the the field to be `X-Concurrent-Job-Limit`, and [stack overflow](https://stackoverflow.com/questions/5258977/are-http-headers-case-sensitive#:~:text=It's%20case%20insensitive%2C%20but%20if,be%20'Content%2DType'.) says  that response headers are case-insensitive but general practice is to keep the first letter capital as shown above.

Comment (NitinBindal-Hevo) [reply src/twitter4jads/util/TwitterAdHttpUtils.java:73]: Will definitely look into this issue

Comment (NitinBindal-Hevo) [reply src/twitter4jads/util/TwitterAdHttpUtils.java]: Done

**Review by adithya-1 — APPROVED** [review]

File: src/twitter4jads/util/TwitterAdHttpUtils.java [hunk 3]
```diff
@@ -64,4 +65,21 @@ public static RateLimitStatusImpl createFromResponseHeaderForCostBasedRateLimit(
 
         return new RateLimitStatusImpl(remainingHits, limit, resetTimeInSeconds, costValue);
     }
-}
+
+    public static JobLimitStatusImpl getJobLimitDataFromResponseHeader(HttpResponse res) {
+        int limit = 0;
+        int remaining = 0;
+
+        String strLimit = res.getResponseHeader("X-Concurrent-Job-Limit");
```
Comment (shashank-hevo) [inline src/twitter4jads/util/TwitterAdHttpUtils.java:73]: Can we use constants for the headers?

**Review by shashank-hevo — COMMENTED** [review]
Have added one minor comment. Apart from that looks good to me.

File: src/twitter4jads/util/TwitterAdHttpUtils.java (hunk 3 above)
Comment (NitinBindal-Hevo) [reply src/twitter4jads/util/TwitterAdHttpUtils.java:73]: Yaa i got this issue earlier as well actually if you see the whole file the same type of style is coded in that object so to keep the code uniform i. have done this like in same file there is a RateLimitStatus object in that same thing is done .

Comment (shashank-hevo) [reply src/twitter4jads/util/TwitterAdHttpUtils.java:73]: Approving this for now, but you can change all the headers to constants in `TwitterAdHttpUtils.java` later

**Review by shashank-hevo — APPROVED** [review]

File: src/twitter4jads/util/TwitterAdHttpUtils.java (hunk 3 above)
Comment (NitinBindal-Hevo) [reply src/twitter4jads/util/TwitterAdHttpUtils.java:73]: OKay

**Pull request merged**